Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -126,7 +124,7 @@ public void OnActivityStart(HttpRequestMessage request)
             if (request != null &&
                 !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString()))
             {
-                this.InjectRequestHeaders(request, this.client.Context.InstrumentationKey);
+                this.InjectRequestHeaders(request, this.configuration.InstrumentationKey);
```

Comment:
the idea is to avoid storing DependencyTelemetry in dictionary between start and stop: all necessary information is available in Stop event. DependencyTelemetry is created there.

The only thing that is needed is instrumentationKey and I assume it is static (common for all requests). Role information is also static.

If you prefer to create telemetry in start and initialize it (and use more memory and CPU) just to fill in static properties that never change, I can do it. But I don;t think it's a great idea.